packages/core/server/admin-api/services/url-pattern.ts

```diff
@@ -175,37 +154,49 @@ export default () => ({
 
     return fields;
   },
-
-
   /**
    * Get all fields from a pattern.
    *
    * @param {string} pattern - The pattern.
    *
    * @returns {array} The fields.\[([\w\d\[\]]+)\]
    */
-  getFieldsFromPattern: (pattern: string) => {
-    const fields = pattern.match(/[[\w\d.]+]/g); // Get all substrings between [] as array.
-    if (!fields) return [];
-    const newFields = fields.map((field) => (/(?<=\[)(.*?)(?=\])/).exec(field)?.[0] ?? ''); // Strip [] from string.
+  getFieldsFromPattern: (patterns: string | string[]) => {
```

Comment:
I would rather be consistent here and require the patterns param to always be an array.